guardian/amigo
# Roles for adding Amazon Open Telemetry agent to the box

## What does this change?

Adds two new roles - `otel-collector-agent-arm` and `otel-collector-agent-x86`.  These install Amazon's Open Telemetry agent to the box, which can be used for e.g. Prometheus metric shipping, AWS X-ray tracing, etc.

Further documentation is in the added README files

## How to test

Perform a build with the role enabled.  Once you start an instance with the resulting build, you should see the agent installed as per the README instructions.

You should be able to fire it up by following the README instructions to provide a configuration file and manually start it.

## How can we measure success?

Able to ship metrics

## Have we considered potential risks?

None, this is strictly opt-in.  The software has been tested by DevX and is AWS supported.
